api/handlers_instances.go

```diff
@@ -495,21 +495,21 @@ func (s *server) CreateAssociationHandler(w http.ResponseWriter, r *http.Request
 	handleResponseOk(w, struct{ AssociationId string }{AssociationId: *out.AssociationDescription.AssociationId})
 }
 
-func (s *server) InstancesTagsHandler(w http.ResponseWriter, r *http.Request) {
+func (s *server) InstanceUpdateHandler(w http.ResponseWriter, r *http.Request) {
 	w = LogWriter{w}
 	vars := mux.Vars(r)
 	account := s.mapAccountNumber(vars["account"])
 	instanceId := vars["id"]
 
-	req := &Ec2ImageUpdateRequest{}
+	req := &Ec2InstanceUpdateRequest{}
 	if err := json.NewDecoder(r.Body).Decode(req); err != nil {
 		msg := fmt.Sprintf("cannot decode body into update image input: %s", err)
 		handleError(w, apierror.New(apierror.ErrBadRequest, msg, err))
 		return
 	}
 
-	if len(req.Tags) == 0 {
-		handleError(w, apierror.New(apierror.ErrBadRequest, "missing required field: tags", nil))
+	if len(req.Tags) == 0 && len(req.InstanceType) == 0 {
+		handleError(w, apierror.New(apierror.ErrBadRequest, "missing required fields", nil))
 		return
 	}
 
```

Comment:
I think you should create and use a separate `instanceUpdatePolicy` here (since you need more permissions than just tags)

Reply:
updated code.